exporter/sematextexporter/writer_test.go

```diff
@@ -0,0 +1,235 @@
+		expectMultipleRequests bool
+	}{
+		{
+			name:                   "default",
+			payloadMaxLines:        10_000,
+			payloadMaxBytes:        10_000_000,
+			expectMultipleRequests: false,
+		},
+		{
+			name:                   "limit-lines",
+			payloadMaxLines:        1,
+			payloadMaxBytes:        10_000_000,
+			expectMultipleRequests: true,
+		},
+		{
+			name:                   "limit-bytes",
+			payloadMaxLines:        10_000,
+			payloadMaxBytes:        1,
+			expectMultipleRequests: true,
+		},
+	} {
+		t.Run(testCase.name, func(t *testing.T) {
+			var httpRequests []*http.Request
+
+			mockHTTPService := httptest.NewServer(http.HandlerFunc(func(_ http.ResponseWriter, r *http.Request) {
+				httpRequests = append(httpRequests, r)
+			}))
+			t.Cleanup(mockHTTPService.Close)
+
+			batch := &sematextHTTPWriterBatch{
+				sematextHTTPWriter: &sematextHTTPWriter{
+					encoderPool: sync.Pool{
+						New: func() any {
+							e := new(lineprotocol.Encoder)
+							e.SetLax(false)
+							e.SetPrecision(lineprotocol.Nanosecond)
+							return e
+						},
+					},
+					httpClient:      &http.Client{},
+					writeURL:        mockHTTPService.URL,
+					payloadMaxLines: testCase.payloadMaxLines,
+					payloadMaxBytes: testCase.payloadMaxBytes,
+					logger:          common.NoopLogger{},
+				},
+			}
+
+			err := batch.EnqueuePoint(context.Background(), "m", map[string]string{"k": "v"}, map[string]any{"f": int64(1)}, time.Unix(1, 0), 0)
+			require.NoError(t, err)
+			err = batch.EnqueuePoint(context.Background(), "m", map[string]string{"k": "v"}, map[string]any{"f": int64(2)}, time.Unix(2, 0), 0)
+			require.NoError(t, err)
+			err = batch.WriteBatch(context.Background())
+			require.NoError(t, err)
+
+			if testCase.expectMultipleRequests {
+				assert.Len(t, httpRequests, 2)
+			} else {
+				assert.Len(t, httpRequests, 1)
+			}
+		})
+	}
+}
+
+// This is the test failing:
+// Error:      	Received unexpected error:
+//
+//	            	Permanent error: line protocol write returned "400 Bad Request" "ERROR: Can't parse line m,k=v,os.host=ADMINs-MacBook-Pro.local,token=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx f=1i 1000000002000 . Invalid timestamp format (must be 19 digit, nanosecond-precision Unix time), found `1000000002000`.; \n"
+//	Test:       	Test_sematextHTTPWriterBatch_EnqueuePoint_emptyTagValue
+func Test_sematextHTTPWriterBatch_EnqueuePoint_emptyTagValue(t *testing.T) {
+	var recordedRequest *http.Request
+	var recordedRequestBody []byte
+
+	// Set up a mock HTTP server to capture requests
+	noopHTTPServer := httptest.NewServer(http.HandlerFunc(func(_ http.ResponseWriter, r *http.Request) {
+
+		if assert.Nil(t, recordedRequest) {
+			var err error
+			recordedRequest = r
+			recordedRequestBody, err = io.ReadAll(r.Body)
+			if err != nil {
+				fmt.Println(err.Error())
+			}
+		}
+	}))
+	t.Cleanup(noopHTTPServer.Close)
+
+	// Set up the time for the point being enqueued
+	nowTime := time.Unix(1628605794, 318000000)
```

Comment:
So we changed the test? I didn't like this. It may lead to have unknown bugs.

Reply:
I can return it as is without any issues. Since the test is an `assert.Equal` test, it will pass as long as the output matches the expected format. The real reason the test was failing was because the endpoint was incorrect